src/lib/y2storage/proposal/autoinst_devices_creator.rb

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env ruby
+#
+# encoding: utf-8
+
+# Copyright (c) [2017] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "y2storage/proposal/partitions_distribution_calculator"
+# TODO: fix distribution calculator to don't require this
+require "y2storage/proposal/lvm_helper"
+require "y2storage/proposal/partition_creator"
+
+module Y2Storage
+  module Proposal
+    # Class to create and reuse devices during the AutoYaST proposal, based
+    # on the information contained in the profile.
+    class AutoinstDevicesCreator
+      include Yast::Logger
+
+      # Constructor
+      #
+      # @param original_graph [Devicegraph] Devicegraph to be used as starting point
+      def initialize(original_graph)
+        @original_graph = original_graph
+      end
+
+      # Devicegraph including all the specified planned devices
+      #
+      # @param planned_devices [Array<Planned::Partition>] Devices to create/reuse
+      # @param disk_names [Array<String>] Disks to consider
+      # @return [Devicegraph] New devicegraph in which all the planned devices have been allocated
+      def devicegraph(planned_devices, disk_names)
```

Comment:
I suggest to at least rename it to `devicegraph_for` otherwise it is quite confusing as I already mention above.
BTW related to discussion on IRC. here it is not that arguments are not modified, but that it `return copy of of original graph` so it is not that argument is not modified, but that devicegraph is related to original one. So it should be mentioned.